sql/src/main/java/org/apache/druid/sql/http/ResultFormat.java

```diff
@@ -133,7 +134,10 @@ public interface Writer extends Closeable
 
     void writeHeader(RelDataType rowType, boolean includeTypes, boolean includeSqlTypes) throws IOException;
 
-    void writeHeaderFromRowSignature(RowSignature rowSignature, boolean includeTypes) throws IOException;
+    default void writeHeaderFromRowSignature(RowSignature rowSignature, boolean includeTypes) throws IOException
```

Comment:
The interface isn't marked as `PublicApi` or `ExtensionPoint`. Is there any reason, we want to add this default behavior on account of it being a no-op?

The callers of the method would call this method indiscriminately since there's no check like ` boolean canWriteHeaderFromRowSignature()`, which would lead to a runtime error, instead of a compilation failure. Any extension or class modifying it should fail to compile, and after consideration, be no-op/throw exception/add implementation, depending on the implementation; than compiling, and throwing at runtime. (Since there's no flag that the callers can invoke to see if the caller can call the method safely or not)

Reply:
Also, a better default implementation would be deferring the call to `writeHeader`, after constructing a `RelDataType` from `rowType`. The types should be omitted since that would be ambiguous (a single `ColumnType` can to multiple `RelDataType`).
However, I am in favor of removing the default implementation altogether.